theano/compile/debugmode.py

```diff
@@ -659,12 +663,25 @@ def get_id_str(obj):
                                print_type=print_type, file=file, order=order,
                                ids=ids, stop_on_name=stop_on_name,
                                prefix_child=new_prefix_child, scan_ops=scan_ops,
-                               profile=profile)
+                               profile=profile,
+                               scan_inner_to_outer_inputs=scan_inner_to_outer_inputs)
 
     else:
-        # this is an input variable
-        id_str = get_id_str(r)
-        print('%s%s %s%s' % (prefix, r, id_str, type_str), file=file)
+        if scan_inner_to_outer_inputs is not None and\
+           r in scan_inner_to_outer_inputs:
+
+            id_str = get_id_str(r)
+            outer_r = scan_inner_to_outer_inputs[r]
+
+            if hasattr(outer_r.owner, 'op'):
+                outer_id_str = get_id_str(outer_r.owner)
+            else:
+                outer_id_str = get_id_str(outer_r)
+            print('%s%s %s%s -> %s' % (prefix, r, id_str, type_str, outer_id_str), file=file)
```

Comment:
An arrow might not be the most evident way to signal that one variable refers in a scan to another outside of the scan. Maybe something like "corresponds to" or else. Or maybe keep the array but print something after the debugprints of the scan function to explain the meaning of the arrow. @nouiz  and @lamblin : any opinion on this?